docs/troubleshooting.md

```diff
@@ -46,6 +48,41 @@ This is typically encountered when you attempt to install BKPR with a DNS zone (
 
 ![Azure SP Conflict](images/azure-sp-conflict.png)
 
+## Troubleshooting BKPR ingress
+
+### Let's Encrypt
+
+BKPR deploys [`cert-manager`](#components.md/cert-manager) Kubernetes add-on to automate the management and issuance of TLS certificates. `cert-manager` delegates on Let's Encrypt to retrieve valid X.509 certificates used to protect designated Kubernetes ingress resources with TLS encryption for HTTP traffic.
+
+A Kubernetes ingress resource is designated to be TLS-terminated at the NGINX controller when the following annotations are present:
+
+```
+Annotations:
+  kubernetes.io/ingress.class:                        nginx
+  kubernetes.io/tls-acme:                             true
+```
+
+`cert-manager` watches for Kubernetes ingress resources. When an ingress resource with the `"kubernetes.io/tls-acme": true` annotation is seen for the first time, `cert-manager` tries to issue an [ACME certificate using HTTP-01 challenge](http://docs.cert-manager.io/en/latest/tutorials/acme/http-validation.html). This works by creating a temporary ingress resource name like `cm-acme-http-solver-${pod_id}` and then triggering the HTTP-01 challenge protocol. This requires that Let's Encrypt is able to resolve the DNS name associated with the Kubernetes ingress resource. On a Kubernetes cluster where BKPR is deployed, this requires that External DNS is functioning properly.
```

Comment:
We use "Ingress" (capitalized) everywhere else.

Reply:
Slight reword: 

It does this by creating a temporary Ingress resource named `cm-acme-http-solver-${pod_id}` and then triggering the HTTP-01 challenge protocol. This requires Let's Encrypt to resolve the DNS name associated with the Kubernetes Ingress resource and this, in turn, requires External DNS to be functioning properly.